core/src/main/java/io/smallrye/openapi/runtime/scanner/spi/AnnotationScanner.java

```diff
@@ -370,11 +377,12 @@ default void processResponse(final AnnotationScannerContext context, final Metho
         for (Type type : methodExceptions) {
             DotName exceptionDotName = type.name();
             if (exceptionAnnotationMap != null && !exceptionAnnotationMap.isEmpty()
-                    && exceptionAnnotationMap.keySet().contains(exceptionDotName)) {
-                AnnotationInstance exMapperApiResponseAnnotation = exceptionAnnotationMap.get(exceptionDotName);
-                if (!this.responseCodeExistInMethodAnnotations(context, exMapperApiResponseAnnotation,
-                        apiResponseAnnotations)) {
-                    addApiReponseFromAnnotation(context, exMapperApiResponseAnnotation, operation);
+                    && exceptionAnnotationMap.containsKey(exceptionDotName)) {
+                for (AnnotationInstance exMapperApiResponseAnnotation : exceptionAnnotationMap.get(exceptionDotName)) {
+                    if (!this.responseCodeExistInMethodAnnotations(context, exMapperApiResponseAnnotation,
+                            methodApiResponseAnnotations)) {
```

Comment:
Here we would allow an exception mapper to override class-level responses, but not method responses. This makes sense I think :+1: